docs/source/checkpoints/dist_ckpt.rst

```diff
@@ -0,0 +1,392 @@
+        """ Save and load using basic strategies wrapped with parallelization strategies. """
+        fully_parallel_save_strategy = FullyParallelSaveStrategyWrapper(base_save_strategy)
+        fully_parallel_load_strategy = FullyParallelLoadStrategyWrapper(base_load_strategy)
+        dist_checkpointing.save(sharded_state_dict, ckpt_dir, fully_parallel_save_strategy)
+        return dist_checkpointing.load(sharded_state_dict, ckpt_dir, fully_parallel_load_strategy)
+
+
+The `dist_checkpointing` package provides default strategies for some sharded backends, so it's enough to specify a tuple `(backend, version)` as a saving strategy.
+Backends and versions are stored in a `metadata.json` file inside the checkpoint so that the loading strategy can be determined automatically (provided that there exists a default loading strategy for a given backend and version).
+
+For "sharded" strategies, currently the backends supported by default are based on `torch.distributed.checkpoint` format (`torch_dist` backend) and Zarr format (`zarr` backend).
+Additionally, as shown in the example above, some wrappers are provided that allow to parallelize the save and load across the whole workload (assuming some data duplication).
+
+For "common" strategies, currently the only supported one is `torch` which saves "common" data into a `common.pt` file.
+
+Optimizers
+==========
+This module gives helper tools to the user to simplify constructing ShardedTensors for optimizer states.
+The ShardedTensors that define local to sharded tensors mapping for model parameters should be reused for optimizer states to avoid code duplication.
+
+To this end, the `dist_checkpointing.optimizers.get_param_id_to_sharded_param_map` function can build a mapping between optimizer params ids and model ShardedTensors.
+This mapping can be used by the `dist_checkpointing.optimizers.optim_state_to_sharding_state` function or application code (for non-standard use cases) to construct optimizer sharded state dict with ShardedTensors.
+This should support most optimizer cases, but some of them might require custom sharded state dict creation.
+A good example is a Distributed Optimizer which flattens the parameters - see `Tensors transformations`_ section for more details.
+
+Note: in order to reuse model SharderTensors to create optimizer ShardedTensors, the model **SharderTensors must wrap model parameters**, not just tensors
+(obtaining a state dict with model parameters can be achieved by passing `keep_vars=True` to the model `state_dict` function).
+Otherwise the correspondence between model ShardedTensors and optimizer states is impossible to recreate.
+This is the reason for introducing ShardedTensorFactories - we have to register the original model parameter as `ShardedTensorFactories.data` and apply any subsequent transformations as a factory function in order to make sure that the same transformation can be applied to the optimizer states.
```

Comment:
To reuse model ShardedTensors for creating optimizer ShardedTensors, it is crucial that the model ShardedTensors directly wrap model parameters, rather than just tensors. You can obtain a state dictionary with model parameters by passing keep_vars=True to the model’s state_dict function. Without this correspondence between model ShardedTensors and optimizer states, recreating the mapping becomes impossible. This is precisely why we introduce ShardedTensorFactories. By registering the original model parameter as ShardedTensorFactories.data and applying any subsequent transformations as a factory function, we ensure that the same transformation can be consistently applied to the optimizer states.